### PR TITLE
 Async version of multiple sequential chat

### LINKED ISSUE #1596: [Issue]: Async version of multiple sequential chat
### Describe the issue

Add an async version of [`initiate_chats`](https://github.com/microsoft/autogen/blob/main/autogen/agentchat/conversable_agent.py#L877).

Multi sequential chats notebook: https://github.com/microsoft/autogen/blob/main/notebook/agentchat_multi_task_chats.ipynb

### Steps to reproduce

_No response_

### Screenshots and logs

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by randombet**
## Why are these changes needed?
This PR add a new  a_initiate_chats interface to add concurrent execution in comparison to initiate_chats interface.
It allows one to pass multiple tasks and their corresponding dedicated agents and execute concurrently. Depending on the prerequisite task(s), the tasks will be solved in a concurrent manner.

## Related issue number 

Closes #1596 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

File: test/agentchat/test_assistant_agent.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
     skip = False or skip_openai
 
 KEY_LOC = "notebook"
-OAI_CONFIG_LIST = "OAI_CONFIG_LIST"
+OAI_CONFIG_LIST = "../OAI_CONFIG_LIST"
```
Comment (sonichi) [inline test/agentchat/test_assistant_agent.py:18]: Don't change this. Change the location of your OAI_CONFIG_LIST file in your local environment.

File: test/agentchat/test_async_chats.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+from test_assistant_agent import KEY_LOC, OAI_CONFIG_LIST
+import pytest
+from conftest import skip_openai
+import autogen
+from typing import Literal
+from typing_extensions import Annotated
+from autogen import initiate_chats
+
+
+@pytest.mark.asyncio
```
Comment (sonichi) [inline test/agentchat/test_async_chats.py:13]: add the skip condition `skip_openai`

File: notebook/agentchat_multi_task_async_chats.ipynb
Comment (sonichi) [inline notebook/agentchat_multi_task_async_chats.ipynb:1]: Add a link to [Examples.md](https://microsoft.github.io/autogen/docs/Examples/)

File: test/agentchat/test_async_chats.py (hunk 2 above)
Comment (randombet) [reply test/agentchat/test_async_chats.py:14]: Added

File: test/agentchat/test_assistant_agent.py (hunk 1 above)
Comment (randombet) [reply test/agentchat/test_assistant_agent.py]: Done

File: notebook/agentchat_multi_task_async_chats.ipynb
Comment (randombet) [reply notebook/agentchat_multi_task_async_chats.ipynb:1]: Done

**Comment by sonichi** [comment]
Could you fix the test error: https://github.com/microsoft/autogen/actions/runs/7964456397/job/21742029123?pr=1724

**Review by qingyun-wu — COMMENTED** [review]
The changes in the PR look great! Thank you @randombet! I think it would be even better if we also add methods such as `add_chat`, `remove_chat` which allow one to add or remove chat in the chat_queue concurrent to `a_initiate_chats`. This could be done in a separate PR!

**Review by qingyun-wu — APPROVED** [review]

**Comment by randombet** [comment]
> The changes in the PR look great! Thank you @randombet! I think it would be even better if we also add methods such as `add_chat`, `remove_chat` which allow one to add or remove chat in the chat_queue concurrent to `a_initiate_chats`. This could be done in a separate PR! 

Cool. Let's do it in another PR

**Pull request merged**